tests/prop.rs

```diff
@@ -147,3 +174,23 @@ fn prove_chain() {
         ],
     );
 }
+
+#[test]
+fn evaluate() {
+    let start = "(| (& F T) (& T F))";
+    let start_expr = Prop.parse_expr(start).unwrap();
+    assert_eq!(Prop::eval(&start_expr), Bool::False);
+}
+
+#[test]
+fn const_fold() {
+    let start = "(| (& F T) (& T F))";
+    let start_expr = Prop.parse_expr(start).unwrap();
+    let (mut eg, _) = EGraph::from_expr(&start_expr);
+    for n in 1..4 {
```

Comment:
Why 4? This should probably assert that you eventually stop folding.

Reply:
I just put the test there to look at the dot files. How do you assert folding stops? Just loop until fix point?

Reply:
I’d unroll the loop and assert the return value of fold_constants (and maybe prune) until it’s 0.

Note that you’ll have to make one of the other fixes first, so that fold_constants actually returns 0.

Reply:
Got it